storage/range.go

```diff
@@ -956,33 +956,33 @@ func (r *Range) applyRaftCommandInBatch(ctx context.Context, index uint64, origi
 
 // getLeaseForGossip tries to obtain a leader lease. Only one of the replicas
 // should gossip; the bool returned indicates whether it's us.
-func (r *Range) getLeaseForGossip(ctx context.Context) (bool, error) {
+func (r *Range) getLeaseForGossip(ctx context.Context) (hasLease bool, err error) {
 	// If no Gossip available (some tests) or range too fresh, noop.
 	if r.rm.Gossip() == nil || !r.isInitialized() {
 		return false, util.Errorf("no gossip or range not initialized")
 	}
-	task := r.rm.Stopper().StartTask()
-	if !task.Ok() {
-		return false, util.Errorf("node is stopping")
-	}
-	defer task.Done()
-
-	timestamp := r.rm.Clock().Now()
-
-	// Check for or obtain the lease, if none active.
-	err := r.redirectOnOrAcquireLeaderLease(tracer.FromCtx(ctx), timestamp)
-	if err != nil {
-		switch e := err.(type) {
-		// NotLeaderError means there is an active lease, leaseRejectedError
-		// means we tried to get one but someone beat us to it.
-		case *proto.NotLeaderError, *proto.LeaseRejectedError:
-		default:
-			// Any other error is worth being logged visibly.
-			log.Warningc(ctx, "could not acquire lease for range gossip: %s", e)
-			return false, err
+	hasLease = true
+	if !r.rm.Stopper().RunTask(func() {
+		timestamp := r.rm.Clock().Now()
+
+		// Check for or obtain the lease, if none active.
+		err = r.redirectOnOrAcquireLeaderLease(tracer.FromCtx(ctx), timestamp)
+		if err != nil {
+			switch e := err.(type) {
+			// NotLeaderError means there is an active lease, leaseRejectedError
+			// means we tried to get one but someone beat us to it.
+			case *proto.NotLeaderError, *proto.LeaseRejectedError:
```

Comment:
@bdarnell  hasLease needs to be set to false here right? I think I broke this. I'll send you a PR.